### PR TITLE
[FS] Cannot deposit funds into sidechains

### PR DESCRIPTION
**Pull request by quantumagi**
Withdrawal transactions forwarding amounts to the federation's sidechain multisig address cause the `ExtractWithdrawalTransaction` to fail due to more than one output containing federation's sidechain multisig address .

This PR:
- Changes `ExtractWithdrawalTransaction` to not recognize such withdrawals as valid.
- Ignores such deposits in `RecordLatestMatureDeposits`.
- Adds a test case.

### CODE REVIEW

**Comment by fassadlr** [comment]
@quantumagi can you fix the merge conflict? Thanks.

**Comment by fassadlr** [comment]
@quantumagi build is broken bru.

**Comment by quantumagi** [comment]
@fassadlr , Yeah some weird goings on. Fixin' it.

File: src/Stratis.Features.FederatedPeg.Tests/CrossChainTransferStoreTests.cs [hunk 1]
```diff
@@ -639,5 +639,48 @@ public async Task AttemptFederationInvalidWithdrawal()
...
+                        BlockHash = 1,
+                        BlockHeight = crossChainTransferStore.NextMatureDepositHeight },
+                    new[] { deposit1 })
+                };
+
+                crossChainTransferStore.RecordLatestMatureDepositsAsync(blockDeposits).GetAwaiter().GetResult();
+
+                Transaction[] transactions = crossChainTransferStore.GetTransactionsByStatusAsync(CrossChainTransferStatus.Partial).GetAwaiter().GetResult().Values.ToArray();
+
+                Assert.Empty(transactions);
```
Comment (codingupastorm) [inline src/Stratis.Features.FederatedPeg.Tests/CrossChainTransferStoreTests.cs]: Just like you do for Partial, check also for Suspended. If it's in there it's a problem...

File: src/Stratis.Features.FederatedPeg.Tests/CrossChainTransferStoreTests.cs [hunk 2]
```diff
@@ -639,5 +639,48 @@ public async Task AttemptFederationInvalidWithdrawal()
...
+            using (ICrossChainTransferStore crossChainTransferStore = this.CreateStore())
+            {
+                crossChainTransferStore.Initialize();
+                crossChainTransferStore.Start();
+
+                TestBase.WaitLoopMessage(() => (this.ChainIndexer.Tip.Height == crossChainTransferStore.TipHashAndHeight.Height, $"ChainIndexer.Height:{this.ChainIndexer.Tip.Height} Store.TipHashHeight:{crossChainTransferStore.TipHashAndHeight.Height}"));
+                Assert.Equal(this.ChainIndexer.Tip.HashBlock, crossChainTransferStore.TipHashAndHeight.HashBlock);
+
+                // Forwarding money already in the multisig address to the muultisig address is ignored.
+                BitcoinAddress address1 = multiSigAddress.RedeemScript.Hash.GetAddress(this.network);
```
Comment (codingupastorm) [inline src/Stratis.Features.FederatedPeg.Tests/CrossChainTransferStoreTests.cs]: Good, also do another transfer (another test is fine) where the opreturndata is a different multisig address.

File: src/Stratis.Features.FederatedPeg.Tests/CrossChainTransferStoreTests.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Features.FederatedPeg.Tests/CrossChainTransferStoreTests.cs]: Done.

File: src/Stratis.Features.FederatedPeg.Tests/CrossChainTransferStoreTests.cs (hunk 2 above)
Comment (quantumagi) [reply src/Stratis.Features.FederatedPeg.Tests/CrossChainTransferStoreTests.cs]: Done. Used the same test to save on test execution time.

**Pull request merged**